src/Simplify.elm

```diff
@@ -6445,10 +6453,110 @@ listSortChecks checkInfo =
     firstThatConstructsJust
         [ \() -> callOnEmptyReturnsEmptyCheck checkInfo.firstArg listCollection checkInfo
         , \() -> callOnSingletonListDoesNotChangeItCheck checkInfo.firstArg checkInfo
+        , \() -> operationDoesNotChangeResultOfOperationCheck checkInfo
         ]
         ()
 
 
+listSortCompositionChecks : CompositionIntoCheckInfo -> Maybe ErrorInfoAndFix
+listSortCompositionChecks checkInfo =
+    operationDoesNotChangeResultOfOperationCompositionCheck { argCount = 1 } checkInfo
+
+
+{-| Condense applying the the same function with equal arguments (except the last one) twice in sequence into one.
+This applies to functions that are equivalent to identity when operating on the result another such function.
+
+Examples of such functions:
+
+  - one argument: `Simplify.expectNaN`, `Review.Rule.providesFixesForModuleRule`, `List.sort`, `List.Extra.unique`, [`AVL.Set.clear`](https://package.elm-lang.org/packages/owanturist/elm-avl-dict/2.1.0/AVL-Set#clear)
+  - two arguments: `List.filter f`, `List.take/drop n`, `List.Extra.filterNot f`, `List.Extra.takeWhile/dropWhile(Right) f`, `List.sortBy f`, `List.sortWith f`, `List.Extra.uniqueBy f`
```

Comment:
`List.drop n >> List.drop n` results in `List.drop (n + n)`, so it's not a good example. (But it could be a new simplification idea), I'll add it to #2.

Reply:
very true. My mind was in `drop/takeWhile` land. Fixed.